core/src/main/java/org/apache/gravitino/storage/relational/mapper/provider/base/TableColumnBaseSQLProvider.java

```diff
@@ -132,4 +132,37 @@ public String selectColumnPOById(@Param("columnId") Long columnId) {
         + " WHERE column_id = #{columnId} AND deleted_at = 0"
         + " ORDER BY table_version DESC LIMIT 1";
   }
+
+  public String listColumnPOsByTableIds(@Param("columnIds") List<Long> columnIds) {
+    return "<script>"
+        + " SELECT c.column_id AS columnId, c.column_name AS columnName,"
+        + " c.column_position AS columnPosition, c.metalake_id AS metalakeId, c.catalog_id AS catalogId,"
+        + " c.schema_id AS schemaId, c.table_id AS tableId,"
+        + " c.table_version AS tableVersion, c.column_type AS columnType,"
+        + " c.column_comment AS columnComment, c.column_nullable AS nullable,"
+        + " c.column_auto_increment AS autoIncrement,"
+        + " c.column_default_value AS defaultValue, c.column_op_type AS columnOpType,"
+        + " c.deleted_at AS deletedAt, c.audit_info AS auditInfo"
+        + " FROM "
+        + TableColumnMapper.COLUMN_TABLE_NAME
+        + " c "
+        + "JOIN ("
+        + "    SELECT column_id, MAX(table_version) AS max_version"
+        + "    FROM "
+        + TableColumnMapper.COLUMN_TABLE_NAME
+        + "    WHERE column_id IN ("
+        + "        <foreach collection='columnIds' item='columnId' separator=','>"
+        + "          #{columnId}"
+        + "        </foreach>"
+        + "    ) AND deleted_at = 0 GROUP BY column_id"
+        + ") latest "
+        + "ON c.column_id = latest.column_id AND c.table_version = latest.max_version"
+        + " WHERE c.column_id IN ("
+        + "<foreach collection='columnIds' item='columnId' separator=','>"
+        + "#{columnId}"
+        + "</foreach>"
+        + ") "
+        + " AND c.deleted_at = 0"
+        + "</script>";
+  }
```

Comment:
Does this SQL work for H2 and PG?

Reply:
Can you add a test to cover this?

Reply:
`TagIT` has tests `tag.associatedObjects()` to cover this changes and It should be work for H2 and PG as the IT has been run with backend `H2` and `PG`.

<img width="1274" alt="image" src="https://github.com/user-attachments/assets/b73440d0-69dc-48ae-b0b4-9fc4248ba618" />